### PR TITLE
refactor: use Fail template tag in SwingSet

### PR DESCRIPTION
**Pull request by erights**
The SwingSet-only portion of https://github.com/Agoric/agoric-sdk/pull/6482

See https://github.com/endojs/endo/pull/1334

### CODE REVIEW

File: packages/SwingSet/src/kernel/vat-loader/manager-factory.js [hunk 1]
```diff
@@ -65,12 +65,11 @@ export function makeVatManagerFactory({
...
-    assert(
-      !bundle || typeof bundle === 'object',
-      `bundle must be object, not a plain string`,
-    );
+    !bundle ||
+      typeof bundle === 'object' ||
+      Fail`bundle must be object, not a plain string`;
     // todo maybe useless
-    assert(!(setup && !enableSetup), X`setup() provided, but not enabled`);
+    !(setup && !enableSetup) || Fail`setup() provided, but not enabled`;
```
Comment (gibson042) [inline packages/SwingSet/src/kernel/vat-loader/manager-factory.js]:
```suggestion
    !setup || enableSetup || Fail`setup() provided, but not enabled`;
```

File: packages/SwingSet/src/lib/id.js [hunk 2]
```diff
@@ -24,10 +24,10 @@ import { assert, details as X } from '@agoric/assert';
 export function insistVatID(s) {
   try {
     assert.typeof(s, 'string', X`not a string`);
-    assert(s.startsWith(`v`), X`does not start with 'v'`);
+    s.startsWith(`v`) || Fail`does not start with 'v'`;
     Nat(BigInt(s.slice(1)));
   } catch (e) {
-    assert.fail(X`'${s} is not a 'vNN'-style VatID: ${e}`);
+    Fail`'${s} is not a 'vNN'-style VatID: ${e}`;
```
Comment (gibson042) [inline packages/SwingSet/src/lib/id.js]:
```suggestion
    Fail`${s} is not a 'vNN'-style VatID: ${e}`;
```

File: packages/SwingSet/src/lib/id.js [hunk 3]
```diff
@@ -56,10 +56,10 @@ export function makeVatID(index) {
 export function insistDeviceID(s) {
   try {
     assert.typeof(s, 'string', X`not a string`);
-    assert(s.startsWith(`d`), X`does not start with 'd'`);
+    s.startsWith(`d`) || Fail`does not start with 'd'`;
     Nat(BigInt(s.slice(1)));
   } catch (e) {
-    assert.fail(X`'${s} is not a 'dNN'-style DeviceID: ${e}`);
+    Fail`'${s} is not a 'dNN'-style DeviceID: ${e}`;
```
Comment (gibson042) [inline packages/SwingSet/src/lib/id.js]:
```suggestion
    Fail`${s} is not a 'dNN'-style DeviceID: ${e}`;
```

File: packages/SwingSet/src/lib/id.js [hunk 4]
```diff
@@ -103,7 +103,7 @@ export function parseVatOrDeviceID(s) {
     type = 'device';
     idSuffix = s.slice(1);
   } else {
-    assert.fail(X`'${s}' is neither a VatID nor a DeviceID`);
+    throw Fail`'${s}' is neither a VatID nor a DeviceID`;
```
Comment (gibson042) [inline packages/SwingSet/src/lib/id.js]:
```suggestion
    throw Fail`${s} is neither a VatID nor a DeviceID`;
```
and a few lines up:
```patch
- assert.typeof(
-   s,
-   'string',
-   X`${s} is not a string, so cannot be a VatID/DeviceID`,
- );
- s = `${s}`;
+ typeof s === 'string' || Fail`${s} is not a string, so cannot be a VatID/DeviceID`;
```

File: packages/SwingSet/src/lib/id.js [hunk 5]
```diff
@@ -24,10 +24,10 @@ import { assert, details as X } from '@agoric/assert';
 export function insistVatID(s) {
   try {
     assert.typeof(s, 'string', X`not a string`);
```
Comment (gibson042) [inline packages/SwingSet/src/lib/id.js]:
```suggestion
    typeof s === 'string' || Fail`not a string`;
```

File: packages/SwingSet/src/lib/id.js [hunk 6]
```diff
@@ -56,10 +56,10 @@ export function makeVatID(index) {
 export function insistDeviceID(s) {
   try {
     assert.typeof(s, 'string', X`not a string`);
```
Comment (gibson042) [inline packages/SwingSet/src/lib/id.js]:
```suggestion
    typeof s === 'string' || Fail`not a string`;
```

**Review by gibson042 — APPROVED** [review]

**Comment by gibson042** [comment]
The test-dapp failure should be fixed by #6630.

File: packages/SwingSet/src/kernel/vat-loader/manager-factory.js (hunk 1 above)
Comment (erights) [reply packages/SwingSet/src/kernel/vat-loader/manager-factory.js]: Done

File: packages/SwingSet/src/lib/id.js (hunk 2 above)
Comment (erights) [reply packages/SwingSet/src/lib/id.js]: Done

File: packages/SwingSet/src/lib/id.js (hunk 3 above)
Comment (erights) [reply packages/SwingSet/src/lib/id.js]: Done

File: packages/SwingSet/src/lib/id.js (hunk 4 above)
Comment (erights) [reply packages/SwingSet/src/lib/id.js]: Done

File: packages/SwingSet/src/lib/id.js (hunk 5 above)
Comment (erights) [reply packages/SwingSet/src/lib/id.js]: Done

File: packages/SwingSet/src/lib/id.js (hunk 6 above)
Comment (erights) [reply packages/SwingSet/src/lib/id.js]: Done

**Pull request merged**